### ISSUE TITLE
[REQUEST] Redundant Save Feature

### ISSUE THREAD

**Issue by RyouMisaki** [notepad-plus-plus/notepad-plus-plus#2098]
Hello people!
First I'd like to thank you for making such a great program that notepad++ is.
I've been using it for some years now, and I never had any big problem with it.... Until 3 days ago :cry: 
I was coding some CSS for my site on my local test server, and I've had done a lot of modifications for some days that I didn't have on the live site.
I don't usually keep backups for css files other than what I have online, but then, while I was making adjustments, I hit Ctrl+S to save and refreshed the page to see it in action, and then poof, notepad++ had crashed while saving, my file was wiped clean. :sob: 
Well, I could redo most of the changes I did before, going with screenshots I had sent (It even got a little better than before), but I fear it might happen again.
I'm not the sort of guy that makes backups constantly, and other people might not even have any backups, so I'm proposing an option for a redundant backup option:

If the user activates this option (Please, on by default!!!), instead of saving the file directly, it would save the file indirectly on a temporary file, like those MS Office lock files.
After the file is ready, it overwrites the file with the temp file, and after it all goes successful , it then deletes the temporary file.

It would take some more time to save a file, maybe half a second on a small file, probably some time on a very big file, but again, it would be optional, so users with very big files that feel they don't need redundancy could skip it, maybe there could even be a shortcut for "Save without redundancy".

This way users files would be protected from sudden deaths while saving, as if for some reason the program crashed, it wouldn't take users files with it.

-----------------------------------------------------------------

TL;DR: Give users an option to save with redundancy, with the file being saved in a temp file then overwritten with the temp content to prevent crashes deleting the file

**Comment by dail8859** [comment]
38ddd32ac51c7a34b6f8f47c9571c45d4dbf9bef was committed a few days ago and *may* be the solution to the error you ran into.

**Comment by RyouMisaki** [comment]
I gave it a look, but there are some things that I still have to learn, so...
Well, it sort of looks like a backup of some sort to prevent data loss, but I didn't fully understand. :sweat_smile: 
How can I get that into my Notepad++? Just downloading it again will do?
(I'm more of a Web Developer than a programmer, help! x. x)

**Comment by dail8859** [comment]
You will have to wait for the next release of Notepad++.  Also keep in mind by default Notepad++ creates backups of files in the appdata directory.

**Comment by RyouMisaki** [comment]
Actually, I didn't know it kept backups, I went to appdata directory and found the backup there... After I redid all that work... :sob: :joy: :laughing: 
Well, then it looks like the save IS already redundant.
It would be nice at least to put some advice in that warning screen, telling users that files might be in %appdata%/Notepad++/backup, I remember only a warning telling me the program crashed

Thanks dail8859!!

**Comment by dail8859** [comment]
Agreed a hint to look in the backup directory might be nice. But since it sounds like the backups is pretty much what you needed I'll go ahead and close the issue.
